app/controllers/dashboard/protocol_merges_controller.rb

```diff
@@ -0,0 +1,139 @@
+# Copyright © 2011-2018 MUSC Foundation for Research Development
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+# disclaimer in the documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products
+# derived from this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
+# BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+# SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+# TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+class Dashboard::ProtocolMergesController < Dashboard::BaseController
+  before_action :authorize_overlord
+  respond_to :json, :html
+
+  def show
+    @user = current_identity
+  end
+
+  def perform_protocol_merge
+    master_protocol = Protocol.where(id: params[:master_protocol_id].to_i).first
+    sub_protocol = Protocol.where(id: params[:sub_protocol_id].to_i).first
+
+    merge_srs = Dashboard::MergeSrs.new()
+    fix_ssr_ids = Dashboard::FixSsrIds.new(master_protocol)
+
+    if (master_protocol == nil) || (sub_protocol == nil)
+      flash[:alert] = 'Protocol(s) not found. Check IDs and try again.'
+    elsif (master_protocol.has_clinical_services? && sub_protocol.has_clinical_services?)
+      flash[:alert] = 'Both protocols have calendars. Only one protocol may have a calendar for merge to work.'
+    else
+      ActiveRecord::Base.transaction do
```

Comment:
This is a lot of content for a controller. It might be better to move this transaction logic to one of the `lib` files you created

Reply:
I'm going to create another service object for this in the next iteration.